node/src/effect/requests.rs

```diff
@@ -911,6 +913,30 @@ impl<I: Display + Debug + Eq, T: Item> Display for FetcherRequest<I, T> {
     }
 }
 
+/// TrieFetcher related requests.
+#[derive(Debug, Serialize)]
+#[must_use]
+pub(crate) struct TrieFetcherRequest<I>
+where
+    I: Debug + Eq,
+{
+    /// The peers to try to fetch from.
+    pub(crate) peers: Vec<I>,
```

Comment:
`FetcherRequest` has only single `peer` to request the data from. Why is `TrieFetcher` different? Can't it reuse the logic for other fetchers?

Reply:
Going with a single peer would make it harder to handle partial failures, I think - as in, we downloaded some chunks successfully, but not others.

I guess we could return some kind of an effect to the reactor that would prompt it to pass another peer, but it seems like it would require more complexity. I'm not sure. I went with this solution, because it seemed more straightforward to me.

Reply:
So what happens if new peers are being connected?  It won't request form them?

This really isn't ideal.  I would suggest either:

  1. Have the trie fetcher component get the peers from the networking component asynchronously somehow or
  2. Just do it in an async function in `operations.rs` where we already have a function that grabs them.